### DISCUSSION TITLE
Documentation on pnpm-lock.yaml sections

### DISCUSSION THREAD

**Discussion by hewelt** [pnpm/pnpm#8792] [Ideas]
👋🏻 Hello there,

We're using `pnpm` in our monorepo and we're happy with it very much 🙏🏻 . Occasionally though, we get a diff on `main`, in `snapshots` section, which contains entries that look like this

```yaml
  follow-redirects@1.15.6:
    optionalDependencies:
      debug: 4.3.7
```

the diff is this

```diff
```yaml
-  follow-redirects@1.15.6:
+  follow-redirects@1.15.6(debug@4.3.7):
    optionalDependencies:
      debug: 4.3.7
```

Now, I don't fully understand why the package in parentheses gets added, especially because removing them doesn't make the pnpm-lock.yaml invalid - `pnpm install --frozen-lockfile` succeeds. 

I assume it may be needed due to the `*` in the `packages` entry and if more than one `debug` version is installed (which is true):

```yaml
  follow-redirects@1.15.6:
   (...)
    peerDependencies:
      debug: '*'
```

but it made me think, that pnpm docs (which are already pretty good) could benefit from having some more documentation on the structure of pnpm-lock.yaml and what's the role of the `snapshots` section and the others. The only issue i found discussing it is this [thread](https://github.com/orgs/pnpm/discussions/8035).

Thanks for doing an amazing work with pnpm 🙇🏻

<img width="300" alt="image" src="https://github.com/user-attachments/assets/d804ad2d-44e4-4cf2-b56e-f67762c420fc">

**Comment by zkochan** [comment]
The lockfile is documented here: https://github.com/pnpm/spec/blob/master/lockfile/9.0.md

In the parentheses those are the resolved peer dependencies. Some info about it is also here: https://pnpm.io/how-peers-are-resolved
